
Fix for #494501

2008-08-16  Robert Millan  <rmh@aybabtu.com>

	* disk/raid.c (grub_raid_init): Handle/report errors set by
	grub_device_iterate().
	* disk/lvm.c (grub_lvm_init): Likewise.

Index: disk/lvm.c
===================================================================
--- disk/lvm.c	(revision 1808)
+++ disk/lvm.c	(revision 1809)
@@ -586,6 +586,12 @@
 GRUB_MOD_INIT(lvm)
 {
   grub_device_iterate (&grub_lvm_scan_device);
+  if (grub_errno)
+    {
+      grub_print_error ();
+      grub_errno = GRUB_ERR_NONE;
+    }
+
   grub_disk_dev_register (&grub_lvm_dev);
 }
 
Index: disk/raid.c
===================================================================
--- disk/raid.c	(revision 1808)
+++ disk/raid.c	(revision 1809)
@@ -580,6 +580,12 @@
 GRUB_MOD_INIT(raid)
 {
   grub_device_iterate (&grub_raid_scan_device);
+  if (grub_errno)
+    {
+      grub_print_error ();
+      grub_errno = GRUB_ERR_NONE;
+    }
+
   grub_disk_dev_register (&grub_raid_dev);
 }
 
